lib/bootstrap/rubygems.rb

```diff
@@ -43,6 +43,25 @@ def self.reset
       end
     end
 
+    ##
+    # Take a plugin name and get the latest versions available in the gem repository.
+    # @param [String] The plugin name
+    # @return [Hash] The collection of registered versions
+    ##
+    def versions(plugin)
+      require "gems"
+      # This patch is necessary to avoid encoding problems when Net:HTTP return stuff in ASCII format, but
+      # consumer libraries, like the YAML parsers expect them to be in UTF-8. As we're using UTF-8 everywhere
+      # and the usage of versions is minimal in our codebase, the patch is done here. If extended usage of this
+      # is done in the feature, more proper fix should be implemented, including the creation of our own lib for
+      # this tasks.
+      ::Gems::Request.module_eval do
```

Comment:
I understand, I will put this in a similar location as the ones in core, it makes sense. On the other side, keep in mind this is a command line tool that run once, so the way this works is actually not that important because the module_eval is going to be done one time and only when the versions function is used. Makes sense to you?